src/compiler/checker.ts

```diff
@@ -8069,7 +8069,8 @@ namespace ts {
             if (containsSameNamedThisProperty(expression.left, expression.right)) {
                 return anyType;
             }
-            const type = resolvedSymbol ? getTypeOfSymbol(resolvedSymbol) : getWidenedLiteralType(checkExpressionCached(expression.right));
+            const isDirectExport = kind === AssignmentDeclarationKind.ExportsProperty && (isPropertyAccessExpression(expression.left) || isElementAccessExpression(expression.left)) && (isModuleExportsAccessExpression(expression.left.expression) || (isIdentifier(expression.left.expression) && isExportsIdentifier(expression.left.expression)));
+            const type = resolvedSymbol ? getTypeOfSymbol(resolvedSymbol) : (isDirectExport ? getRegularTypeOfLiteralType : getWidenedLiteralType)(checkExpressionCached(expression.right));

```

Comment:
This is easier to read flat, even at the cost of some repitition.

```suggestion
            const type = resolvedSymbol ? getTypeOfSymbol(resolvedSymbol) 
                : isDirectExport ? getRegularTypeOfLiteralType(checkExpressionCached(expression.right)) 
                : getWidenedLiteralType(checkExpressionCached(expression.right));
```